src/nccl_ofi_rdma.cpp

```diff
@@ -5918,34 +5688,39 @@ static int send(nccl_net_ofi_send_comm_t *send_comm, void *data, size_t size, in
 		eager = true;
 	}
 
+	msg_seq_num = s_comm->next_msg_seq_num;
+	slot = msg_seq_num % NCCL_OFI_MAX_REQUESTS;
+
+	if (!eager) {
```

Comment:
You changed behaviors here in a fairly significant way.

Previously, we looked for a match and only sent eager if the control hasn't arrived.  Now we're enabling eager even when the control message has arrived.  Given the high cost of an eager message on the receiver, this isn't what we want to do.